workflows/argo-events/workflowtemplates/baremetal-flavor.yaml

```diff
@@ -0,0 +1,133 @@
+spec:
+  serviceAccountName: workflow
+  entrypoint: flavor-check
+  arguments:
+    parameters:
+      - name: flavor_name
+        value: "gp2.small"
+      - name: flavor_properties
+        value: "--property resources:CUSTOM_BAREMETAL_GP2SMALL=1 --property resources:VCPU=0 --property resources:MEMORY_MB=0 --property resources:DISK_GB=0"
+  templates:
+    - name: flavor-check
+      steps:
+        - - name: check-flavor
+            template: check-flavor
+        - - name: create-flavor
+            template: create-flavor
+            arguments:
+              parameters:
+                - name: flavor_properties
+                  value: "{{workflow.parameters.flavor_properties}}"
+            when: "{{steps.check-flavor.outputs.parameters.flavor_exists}} == false"
+        - - name: set-flavor
+            template: set-flavor
+            arguments:
+              parameters:
+                - name: flavor_properties
+                  value: "{{workflow.parameters.flavor_properties}}"
+            when: "{{steps.check-flavor.outputs.parameters.flavor_exists}} == true"
+        - - name: show-flavor
+            template: show-flavor
+
+    - name: show-flavor
+      container:
+        image: ghcr.io/rackerlabs/understack/openstack-client:2024.2-ubuntu_jammy
+        command:
+          - openstack
+        args:
+          - flavor
+          - show
+          - "{{workflow.parameters.flavor_name}}"
+        env:
+          - name: OS_CLOUD
+            value: understack
+        volumeMounts:
+          - mountPath: /etc/openstack
+            name: openstack-svc-acct
+            readOnly: true
+
+    - name: check-flavor
+      container:
+        image: ghcr.io/rackerlabs/understack/openstack-client:2024.2-ubuntu_jammy
+        command:
+          - sh
+          - -c
+          - |
+            echo "Checking if {{workflow.parameters.flavor_name}} flavor exists..."
+            flavor_check=$(openstack --os-cloud understack flavor show {{workflow.parameters.flavor_name}} 2>/dev/null | head -1)
+            if [ -n "$flavor_check" ]; then
+              echo -n "true" > /tmp/flavor_result.txt
+            else
+              echo -n "false" > /tmp/flavor_result.txt
+            fi
+        env:
+          - name: OS_CLOUD
+            value: understack
+        volumeMounts:
+          - mountPath: /etc/openstack
+            name: openstack-svc-acct
+            readOnly: true
+      outputs:
+        parameters:
+          - name: flavor_exists
+            valueFrom:
+              default: "false"
+              path: /tmp/flavor_result.txt
+
+    - name: create-flavor
+      inputs:
+        parameters:
+          - name: flavor_properties
+      container:
+        image: ghcr.io/rackerlabs/understack/openstack-client:2024.2-ubuntu_jammy
+        command:
+          - sh
+          - -c
+          - |
+            eval "openstack flavor create '{{workflow.parameters.flavor_name}}' {{inputs.parameters.flavor_properties}}"
```

Comment:
Not understanding why we need the eval?  Cause the parameters are passed as 1 blob?

https://argo-workflows.readthedocs.io/en/latest/walk-through/scripts-and-results/ might be worth using the script syntax instead of the command syntax then.